### ISSUE TITLE
F1-score very low

### ISSUE THREAD

**Issue by yassmine-lam** [billpku/NLP_In_Action#13]
Hi,

Firstly, thank u for this great repo it was very helpful to me. However, I have a question concerning the evaluation metrics. I adapted this code to another task and data, and I got a very good accuracy (94.xx%) but a very low f1-score (58.xx%). Is that normal plz? Do u have any explanation?

Thank u in advance

**Comment by billpku** [comment]
> Hi,
> 
> Firstly, thank u for this great repo it was very helpful to me. However, I have a question concerning the evaluation metrics. I adapted this code to another task and data, and I got a very good accuracy (94.xx%) but a very low f1-score (58.xx%). Is that normal plz? Do u have any explanation?
> 
> Thank u in advance


Hi Yass，

Sorry for the late reply.
Your are right, we should use suitable metric for specific NLP task，normally：

1.For token classification task，like：NER, extract question answering... the higher F1 score, the better
2.For sentence classification task, like: text classification, the higher accuracy score,the better(not overfitting)

It's normal that, in NER task, you get a hight accuracy score but a low F1 score. Low F1 score indicate that your model not predict well. You can fine tune your model for more epochs or get more labeled data for training. Since NER can be treated as a multi-class classification process, the token label data is very unbalance(think about the distribution of label 'B','O','ORG'.... ), so the accuracy score cannot represent model performance.

For text classification task, low F1 score means that your model is not robust. You can try to balance your training and test labeled data distribution.

Keep in mind that, sometime, the metric may not truly represent your model well only because your training or test data is very unbalance.

**Comment by yassmine-lam** [comment]
Hi,

Thank u very much for ur detailed answer. U made it very clear to me.
